### PR TITLE
Fix transition fail on tailsitter

### PR DESCRIPTION
**Pull request by xdwgood**
**Describe problem solved by this pull request**
To fix: https://github.com/PX4/Firmware/issues/15037

**Describe your solution**
ensure input quaternions are exactly normalized because acosf(1.00001) == NaN

**Test data / coverage**
Test tomorrow

**Additional context**
@sfuhrer  Is my solution the best? Or should we start with `acosf` functions?

### CODE REVIEW

**Comment by xdwgood** [comment]
Ground test log : https://logs.px4.io/plot_app?log=fe37c1fa-2c53-48d1-8efa-7487e60b44a6

@sfuhrer  I conducted at least 50 tests on the ground and the problem did not appear.

Good to merge? :smiley:

**Comment by sfuhrer** [comment]
> Ground test log : https://logs.px4.io/plot_app?log=fe37c1fa-2c53-48d1-8efa-7487e60b44a6
> 
> @sfuhrer I conducted at least 50 tests on the ground and the problem did not appear.
> 
> Good to merge? smiley

Cool! Yes I don't see a reason why not :+1:

**Comment by xdwgood** [comment]
@sfuhrer  done here : https://github.com/PX4/Firmware/pull/15352/commits/7cd4bfd28b86fa5f6a8d3de30c9be28581ec28de :smiley:

File: src/modules/sensors/vehicle_imu/VehicleIMU.cpp [hunk 1]
```diff
@@ -262,7 +262,7 @@ void VehicleIMU::Run()
 
 			// rotate sensor clip counts into vehicle body frame
 			const Vector3f clipping{_accel_calibration.getBoardRotation() *
-				Vector3f{(float)accel.clip_counter[0], (float)accel.clip_counter[1], (float)accel.clip_counter[2]}};
+						Vector3f{(float)accel.clip_counter[0], (float)accel.clip_counter[1], (float)accel.clip_counter[2]}};
```
Comment (sfuhrer) [inline src/modules/sensors/vehicle_imu/VehicleIMU.cpp:265]: @xdwgood it still complains, because of this change here. It's a `make format` misconfiguration, I actually also have it on my setup. So just revert the changes in here, and see that your PR only touches tailsitter.cpp.

Comment (xdwgood) [reply src/modules/sensors/vehicle_imu/VehicleIMU.cpp:265]: I tried to restore, but when I commit it, it showed that the check failed... Can you help me submit a change? Thank you

**Comment by xdwgood** [comment]
@sfuhrer  thanks for your help :heart:  :smiley:
